tests/topotests/ospf_flexalgo_topo1/test_ospf_flexalgo_topo1.py

```diff
@@ -0,0 +1,1963 @@
+            "step1/show_ip_ospf_database_opaque.ref",
+            check_for_mismatch=True
+        )
+
+    # Skip if previous fatal error condition is raised
+    if tgen.routers_have_failure():
+        pytest.skip(tgen.errors)
+
+    #
+    # Action(s):
+    # -Add back the original flags to flexible-algo 128 on rt1
+    #
+    # Expected changes:
+    # -rt1 should update the Flex-Algo with corresponding flags. 
+    # -rt1 MUST advertise the FAD TLV for 128 with original flags
+    #  included in the Flags SubTLV.
+    #
+    step("Try adding original exclude srlg for flex-algo with algorithm Id 128 on rt1")
+    raw_config = {
+        "rt1": {
+            "raw_config": [
+                "router ospf",
+                "flexible-algorithm 128 flags 10"
+            ]
+        }
+    }
+
+    result = apply_raw_config(tgen, raw_config)
+    assert result is True, "Testcase {} : Failed to add flags of Flex-Algo 128 on rt1 again!".format(tc_name)
+
+    # Debug (to be used to debug testcase failure)
+    # for rname in ["rt1", "rt2", "rt3", "rt4"]:
+    #     print_cmd_result("rt4", "show ip ospf database opaque-area")
+
+    for rname in ["rt1", "rt2", "rt3", "rt4"]:
+        router_compare_text_output(
+            rname, "show ip ospf router-info flexible-algorithms",
+            "step1/show_ip_ospf_flexalgo.ref"
+        )
+        router_compare_json_output(
+            rname, "show ip ospf database opaque-area json",
+            "step1/show_ip_ospf_database_opaque.ref"
+        )
+
+    write_test_footer(tc_name)
+
+# Memory leak test template
+def test_memory_leak():
+    "Run the memory leak test and report results."
+    tgen = get_topogen()
+    if not tgen.is_memleak_enabled():
+        pytest.skip("Memory leak test/report is disabled")
+
+    tgen.report_memory_leaks()
+
+
+if __name__ == "__main__":
+    args = ["-s"] + sys.argv[1:]
+    sys.exit(pytest.main(args))
+
+#
+# Step 4
+#
+def test_flex_algo_delete_prefix_advt_metric_step16(request):
+    tgen = get_topogen()
+
+    tc_name = request.node.name
+
+    #
+    # Action(s):
+    # -Delete the prefix-advertise-metric of flexible-algo 128 on rt4
+    #
+    # Expected changes:
+    # -rt4 should remove Ext-Prefix LSA and Ext-IASBR LSA for the Flex Algo 128.
+    #
+
+    # Skip if previous fatal error condition is raised
+    if tgen.routers_have_failure():
+        pytest.skip(tgen.errors)
+
+    step("Try deleting prefix-advertise-metric of flex-algo with algorithm Id 128 on rt4")
+    raw_config = {
+        "rt4": {
+            "raw_config": [
+                "router ospf",
+                "no flexible-algorithm 128 prefix-advertise-metric"
+            ]
+        }
+    }
+
+    result = apply_raw_config(tgen, raw_config)
+    assert result is True, "Testcase {} : Failed to delete prefix-advertise-metric of Flex-Algo 128 on rt4!".format(tc_name)
+
+    # LSA deletion may take upto 360 seconds.
+    topotest.sleep(360)
```

Comment:
This is static sleep, and that too 6 mins. Please check if this can be avoided.

Reply:
We need to get the hidden CLI for bringing down the timer for flushing LSAs first for this test case.